src/check/runner/DecorateProperty.ts

```diff
@@ -22,5 +24,12 @@ export function decorateProperty<Ts>(
     prop = new SkipAfterProperty(prop, Date.now, qParams.skipAllAfterTimeLimit, false);
   if (qParams.interruptAfterTimeLimit != null)
     prop = new SkipAfterProperty(prop, Date.now, qParams.interruptAfterTimeLimit, true);
+  if (qParams.skipEqualValues) {
+    if (qParams.valuesComparator) {
```

Comment:
Code could always pass in that path and let the constructor deal with defaulting the comparator if needed